### ISSUE TITLE
Styles and scripts give 404 when deployed unless the default configuration is repeated (OSOE-877)

### ISSUE THREAD

**Issue by Piedone** [Lombiq/NodeJs-Extensions#104]
Supposedly, this is the default config for `styles` and `scripts`, so you shouldn't have to add it to your _packages.json_:

```js
{
...
  "nodejsExtensions": {
    "styles": {
      "source": "Assets/Styles",
      "target": "wwwroot/css"
    },
    "scripts": {
      "source": "Assets/Scripts",
      "target": "wwwroot/js"
    }
  }
}
```

However, if you omit this, then the files will give a 404 when deployed (to an Azure Web App, if that matters). The files are properly built otherwise though, locally from VS they still work, as well as when built and run with the .NET CLI during UI testing in CI workflows.

Specifying these defaults shouldn't be necessary.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-877)

**Comment by sarahelsaig** [comment]
I'm not sure how to test this, but would deployment work even if you only provide this much?

```js
{
...
  "nodejsExtensions": {
    "styles": { },
    "scripts": { }
  }
}
```

I'm certain that _if_ the style/script build starts, then the config will correctly get the default values in [get-config.js](https://github.com/Lombiq/NodeJs-Extensions/blob/7e1b9ffd68a36ad51cb914a7c9c5d07ef1ad18f7/Lombiq.NodeJs.Extensions/scripts/get-config.js#L47). So I get a sense that maybe the build only triggers if there is a config?

**Comment by Piedone** [comment]
I've sent you a PM on how you can test it with one of our apps (I can't really test it right now) but perhaps you're on the right track.

**Comment by sarahelsaig** [comment]
Where? I don't see it.

**Comment by Piedone** [comment]
I'll give you a hint: it starts with "Teams" and ends with "".

**Comment by sarahelsaig** [comment]
Yes, obviously that's where I first checked. I haven't gotten a PM in Teams from you in a long while

**Comment by Piedone** [comment]
Just check the project channel :). Fair enough, not private, just more private than this here.

**Comment by sarahelsaig** [comment]
Doesn't work, unfortunately. (I get error `Invalid group: {}: source must be a string, target must be a string.`) The defaults are only filled in if you completely omit the (styles, scripts) property in `nodejsExtensions`. It still may be because of what I wrote above, but I don't have an easy workaround to test with.

**Comment by Piedone** [comment]
I see, thank you.

**Comment by Piedone** [comment]
Since the project is now deprecated, we won't do this.
